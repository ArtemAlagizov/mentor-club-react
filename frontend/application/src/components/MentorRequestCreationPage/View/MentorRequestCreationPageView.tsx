import React from "react";
import translator, { scopedTranslate } from "../../../config/i18next";
import { Styled } from "../Styled";
import { Link } from "react-router-dom";
import { IMentorRequestCreationPageViewProps } from "./IMentorRequestCreationPageViewProps";
import { Button, Spin } from "antd";
import FieldInputView from "../../FieldTextInput/View/FieldInputView";
import FieldDropdown from "../../FieldDropdown/Container/FieldDropdown";
import { includes, isNil, keys } from "ramda";
import Card from "../../Card/Container/Card";

const translateMentorRequest = scopedTranslate(
  "mentorRequestCreationPage.mentorRequest"
);

const MentorRequestCreationPageView: React.FC<IMentorRequestCreationPageViewProps> = ({
  mentors,
  setSelectedMentor,
  selectedMentor,
  mentorRequestDescription,
  updateMentorRequestDescription,
  isLoading,
  createMentorRequest
}) => {
  return (
    <Styled.Container>
      <Styled.TopHeaderContainer>
        <Styled.TopHeader>
          {translator.t("mentorRequestCreationPage.title")}
        </Styled.TopHeader>
        <Link to="/mentor-list">
          <Button
            type="ghost"
            style={{
              borderRadius: "0",
              width: "100%",
              marginRight: "5px",
              color: "white"
            }}
            data-id="new-mentor-request-button"
          >
            {translator.t("mentorRequestCreationPage.backToMentorList")}
          </Button>
        </Link>
      </Styled.TopHeaderContainer>

      {isLoading && (
        <Spin>
          <Styled.SpinnerBoundaries />
        </Spin>
      )}

      {!isLoading && (
        <Styled.FormAndButtonContainer>
          <Styled.FormContainer>
            <FieldInputView
              labelText={`${translateMentorRequest("description")}`}
              value={mentorRequestDescription}
              onChange={updateMentorRequestDescription}
              isTextArea={true}
              numberOfLinesForTextArea={4}
            />
            <FieldDropdown
              labelText={`${translateMentorRequest("mentor")}`}
              optionValues={
                isNil(mentors) ? [] : mentors.map(mentor => mentor.userName)
              }
              value={!isNil(selectedMentor) ? selectedMentor.userName : ""}
              onChange={(value: string) => {
                const mentor = isNil(mentors)
                  ? null
                  : mentors.find(mentor => mentor.userName === value);

                if (!isNil(mentor)) {
                  setSelectedMentor(mentor);
                }
              }}
            />
            {!isNil(selectedMentor) && (
              <Styled.CardContainer key={selectedMentor.userId}>
                <Card
                  cardTitle={selectedMentor.userName}
                  cardDataItems={keys(selectedMentor)
                    .filter(key => !includes(key, ["userId", "userRole"]))
                    .sort()
                    .reverse()
                    .map(key => ({
                      label: translateMentorRequest(`mentorProfile.${key}`),
                      valueContainer: {
                        value:
                          key === "userPictureThumbnailSrc" ? (
                            <Styled.UserImage
                              data-id="profile-image"
                              src={selectedMentor.userPictureThumbnailSrc}
                            />
                          ) : (
                            selectedMentor[key]
                          ),
                        customClass:
                          key !== "userPictureThumbnailSrc"
                            ? "card-data-item-text"
                            : ""
                      }
                    }))}
                />
              </Styled.CardContainer>
            )}
          </Styled.FormContainer>
          <Styled.ButtonContainer>
            <Styled.NavigationButton
              type="primary"
              data-id="run-task-button"
              onClick={createMentorRequest}
            >
              {translateMentorRequest("sendRequest")}
            </Styled.NavigationButton>
          </Styled.ButtonContainer>
        </Styled.FormAndButtonContainer>
      )}
    </Styled.Container>
  );
};

export default MentorRequestCreationPageView;
