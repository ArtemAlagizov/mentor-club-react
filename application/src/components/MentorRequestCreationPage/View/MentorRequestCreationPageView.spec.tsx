import React from "react";
import { shallow } from "enzyme";
import MentorRequestCreationPageView from "./MentorRequestCreationPageView";

describe("Given MentorRequestResultPageView", () => {
  describe("when rendering", () => {
    it("should have MentorRequestResultPageView matching snapshot", () => {
      const component = shallow(
        <MentorRequestCreationPageView
          isLoading={false}
          selectedMentor={null}
          createMentorRequest={jest.fn()}
          mentorRequestDescription={""}
          mentors={[]}
          setSelectedMentor={jest.fn()}
          updateMentorRequestDescription={jest.fn()}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
