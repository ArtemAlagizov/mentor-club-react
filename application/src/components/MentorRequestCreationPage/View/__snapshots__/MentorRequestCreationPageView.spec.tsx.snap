// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given MentorRequestResultPageView when rendering should have MentorRequestResultPageView matching snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      New mentor request
    </styled.div>
    <Link
      to="/mentor-requests/new"
    >
      <Button
        block={false}
        data-id="new-mentor-request-button"
        ghost={false}
        htmlType="button"
        loading={false}
        style={
          Object {
            "borderRadius": "0",
            "color": "white",
            "marginRight": "5px",
            "width": "100%",
          }
        }
        type="ghost"
      >
        Back to mentor list
      </Button>
    </Link>
  </styled.div>
  <styled.div>
    <styled.div>
      <FieldInputView
        isTextArea={true}
        labelText="Description"
        numberOfLinesForTextArea={4}
        onChange={[MockFunction]}
        value=""
      />
      <FieldDropdown
        labelText="Mentor"
        onChange={[Function]}
        optionValues={Array []}
        value=""
      />
    </styled.div>
    <styled.div>
      <Styled(Button)
        data-id="run-task-button"
        onClick={[MockFunction]}
        type="primary"
      >
        Send request
      </Styled(Button)>
    </styled.div>
  </styled.div>
</styled.div>
`;
